src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,137 @@
+package duke;
+
+import java.io.IOException;
+import java.time.format.DateTimeParseException;
+
+import duke.command.Command;
+import duke.command.ListCommand;
+import duke.exception.DukeBadInputException;
+
+/**
+ * A task manager that initialise a quack instance
+ */
+public class Duke {
+
+    /**
+     * Instance handling all the user interface
+     */
+    private Ui ui;
+    /**
+     * Instance handling the tasks state
+     */
+    private TaskList taskList;
+    /**
+     * Path to the storage, default is ./data/data.txt
+     */
+    private Storage storage;
+
+    /**
+     * Construct a new Duke object which uses filePath as the storage
+     *
+     * @param filePath - path to the storage file
+     */
+    public Duke(String filePath) {
```

Comment:
Hi Ming Yuan,
I really like the way you have structured your app and it shows your understanding of what the ip requires. 
I have a nit tho. I noticed that the `filepath` argument in Duke constructor is used for initialising the storage. Perhaps it would be better to name it as `storageFilepath`? This would help facilitate possible future features where Duke needs to read from multiple files, and clarify which filepath belongs to which feature.
Overall, good job for ip. Jiayous!
Best,
Wei Hong